### PR TITLE
λ to download fulfilment files to S3

### PR DESCRIPTION
**Pull request by AWare**
Also sets up flow and makes standard a bit pickier.

### CODE REVIEW

File: src/config.js [hunk 1]
```diff
@@ -3,9 +3,10 @@ import NamedError from './NamedError'
...
 export function fetchConfig () {
+  console.log('Fetching configuration file from S3.')
   return new Promise((resolve, reject) => {
-    let stage = process.env.Stage
-    if (stage !== 'CODE' && stage !== 'PROD') {
+    let stage: ?string = process.env.Stage
+    if (stage == null || !['CODE', 'PROD'].includes(stage)) {
       reject(new Error(`invalid stage: ${stage}, please fix Stage env variable`))
       return
     }
```
Comment (pvighi) [inline src/config.js]: just a comment for later: maybe we could put this code somewhere shared as most lambdas need to know the stage

File: src/salesforce_downloader.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      return request.get({ uri: `${j.instance_url}${endpoint}`, headers: { 'Authorization': `Bearer ${j.access_token}` } })
+    },
+    getp: async function (endpoint: string) {
+      return rp.get({ uri: `${j.instance_url}${endpoint}`, headers: { 'Authorization': `Bearer ${j.access_token}` } })
+    }
+  }
+}
+
+async function q () {
+  const prefix = 'CODE/salesforce_output/'
```
Comment (pvighi) [inline src/salesforce_downloader.js]: shouldn't this be `${config.stage}/salesforce_output` ?

File: src/fetcher.js [hunk 3]
```diff
@@ -41,7 +41,7 @@ function fetchFile (batch, deliveryDate, config) {
 }
 function uploadFile (fileData, config) {
   let promise = new Promise((resolve, reject) => {
-    let savePath = `${config.stage}/zuoraExport/${fileData.fileName}`
+    let savePath = `${config.zuora.apistage}/zuoraExport/${fileData.fileName}`
```
Comment (pvighi) [inline src/fetcher.js]: i think this should have remained as it was

File: src/fetcher.js [hunk 4]
```diff
@@ -12,10 +12,10 @@ function fetchFile (batch, deliveryDate, config) {
     console.log(`fetching file from zuora with id ${batch.fileId}`)
     const options = {
       method: 'GET',
-      uri: `${config.url}/apps/api/batch-query/file/${batch.fileId}`,
+      uri: `${config.zuora.apiurl}/apps/api/batch-query/file/${batch.fileId}`,
```
Comment (pvighi) [inline src/fetcher.js]: shouldn't this be config.zuora.api.url instead of config.zuora.apiurl ? and the same for all the other times we read the config

Comment (AWare) [inline src/fetcher.js]: I have no idea how I let this slip through :(

**Comment by pvighi** [comment]
I have not run it but looks ok 👍

**Comment by pvighi** [comment]
👍

**Pull request merged**